KenticoOnboardingApplication/test/KenticoOnboardingApplication.Api.Tests/Controllers/ListControllerTests.cs

```diff
@@ -0,0 +1,297 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+using KenticoOnboardingApplication.Api.Controllers;
+using KenticoOnboardingApplication.Contracts.Helpers;
+using KenticoOnboardingApplication.Contracts.Models;
+using KenticoOnboardingApplication.Contracts.Repositories;
+using KenticoOnboardingApplication.Contracts.Services;
+using KenticoOnboardingApplication.Contracts.Services.Wrappers;
+using KenticoOnboardingApplication.Tests.Base;
+using KenticoOnboardingApplication.Tests.Base.Factories;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace KenticoOnboardingApplication.Api.Tests.Controllers
+{
+    [TestFixture]
+    public class ListControllerTests
+    {
+        private ListController _controller;
+        private IListRepository _repository;
+        private IUrlLocator _urlLocator;
+        private IGetItemService _getItemService;
+        private ICreateItemService _createItemService;
+        private IUpdateItemService _updateItemService;
+
+        private static readonly string[] s_ids =
+        {
+            "00000000-0000-0000-0000-000000000001",
+            "00000000-0000-0000-0000-000000000002",
+            "00000000-0000-0000-0000-000000000003"
+        };
+
+        private static readonly string[] s_texts =
+        {
+            "Learn C#",
+            "Create dummy controller",
+            "Connect JS and TS"
+        };
+
+        private static IEnumerable<(Item, string)> PostTestInvalidItems() =>
```

Comment:
It would greatly improve readability if you named the tuple parts, e.g.:
```suggestion
        private static IEnumerable<(Item errorneousItem, string responseValidationKey)> PostTestInvalidItems() =>
```